File: app/src/main/java/com/jbrunton/mymovies/models/Movie.java
package com.jbrunton.mymovies.models;

import com.google.auto.value.AutoValue;

import org.joda.time.LocalDate;

import java.util.Optional;

@AutoValue
public abstract class Movie {

    public static Movie create(String id, String title, String posterPath, LocalDate releaseDate, String rating, String overview) {
        return new AutoValue_Movie(id, title, posterPath, Optional.ofNullable(releaseDate), rating, Optional.ofNullable(overview));
    }

    public abstract String id();
    public abstract String title();
    public abstract String posterPath();
    public abstract Optional<LocalDate> getReleaseDate();
    public abstract String rating();
    public abstract Optional<String> overview();
}

File: entities/src/main/java/com/jbrunton/mymovies/entities/models/Movie.kt
package com.jbrunton.mymovies.entities.models

import org.joda.time.LocalDate

data class Movie(
    val id: String,
    val title: String,
    val posterUrl: String?,
    val backdropUrl: String?,
    val releaseDate: LocalDate?,
    val rating: String,
    val overview: String?) {
    companion object {
        val emptyMovie = Movie(
                id = "",
                title = "",
                posterUrl = null,
                backdropUrl = null,
                releaseDate = null,
                rating = "",
                overview = null
        )
    }
}
